### PR TITLE
mitigate potential hook race condition and close open handles

### PR DESCRIPTION
**Pull request by therightstuff**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -6,8 +6,13 @@
     "declaration": true,
     "moduleResolution": "node",
     "experimentalDecorators": true,
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "lib": [
+      "dom",
+      "dom.iterable",
```
Comment (doriaviram) [inline tsconfig.json:12]: dom ?

Comment (therightstuff) [reply tsconfig.json:12]: not sure, but i thought it was a good idea to use the same libraries as the integration-tests 🤷

**Review by YonatanHunger — APPROVED** [review]

**Review by nirLumigo — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -44,9 +43,11 @@
     "@types/aws-lambda": "^8.10.96",
     "@types/jest": "^26.0.23",
     "@types/utf8": "^3.0.0",
+    "@typescript-eslint/eslint-plugin": "^5.37.0",
+    "@typescript-eslint/parser": "^5.37.0",
     "aws-sdk": "^2.927.0",
     "axios-mock-adapter": "^1.19.0",
-    "babel-eslint": "^10.1.0",
```
Comment (saartochner) [inline package.json:49]: why did you remove `babel-eslint`?

**Review by saartochner — APPROVED** [review]
LGTM

File: package.json (hunk 2 above)
Comment (therightstuff) [reply package.json:49]: for coherence and consistency, we're now using a typescript- and prettify-compatible linter in both lumigo-node and the integration-tests repositories
